### ISSUE TITLE
PyCharm incorrectly shows type mismatch when using Clone

### ISSUE THREAD

**Issue by Legorooj** [intellij-rust/intellij-rust#6854]
## Environment

* **IntelliJ Rust plugin version:** 0.3.141.3674-203
* **Rust toolchain version:** nightly
* **IDE name and version:** PyCharm 2020.3.3
* **Operating system:** Windows 10

## Problem description

PyCharm raises E0308: expected `Example`, found `&Example`

## Steps to reproduce

Place the following code inside a rust file. 

```rust
#[derive(Clone)]
struct Example {
    pub value: i8
}

struct OtherExample {
    pub example: Example
}

fn mutate(arg: &Example) -> OtherExample {
    OtherExample {
        example: arg.clone()
    }
}

fn main() {
    mutate(
        &Example {
            value: 1
        }
    );
    println!("all good");
}
```
PyCharm should complain. However `cargo check`/`rustc` **do not**.

Screenshot of the error:
![2021-02-16 19_20_41-Project – tests rs  polycalc](https://user-images.githubusercontent.com/50370070/108036305-29907f00-708c-11eb-9983-e2dc659a6d53.png)

**Comment by ortem** [comment]
@Legorooj Thank you for reporting!

Please try executing `Re-Expand All Rust Macros` action, and then `Invalidate Caches` action if the first one doesn't help. Both actions can be invoked via `Ctrl+Shift+A` menu. Feel free to provide any further information.

**Comment by Legorooj** [comment]
Tried both, neither work.

**Comment by Maksim20023** [comment]
Hi! I'm closing this ticket because the issue does not reproduce in the current environment (PyCharm 2023.1.4 Build #PY-231.9225.15, Rust plugin 0.4.199.5415-232). If this issue still reproduces for you, please reopen the ticket.
